api/src/main/java/org/eclipse/microprofile/openapi/models/Reference.java

```diff
@@ -64,6 +64,9 @@
      * @param ref a reference to a T object in the components in this OpenAPI document
      * @return the current instance
      **/
-    T ref(String ref);
+    default T ref(String ref) {
+        setRef(ref);
+        return (T) this;
```

Comment:
Here I am not sure about this, because it generate this warning:

> Type safety: Unchecked cast from Reference<T> to T

Should I add a `@SuppressWarnings("unchecked")` to the method?

An alternative is to do it like that:

```java
    default T ref(String ref) {
        setRef(ref);
        @SuppressWarnings("unchecked")
        T t = (T) this;
        return t;
    }
```